pandas/core/arrays/categorical.py

```diff
@@ -1709,23 +1709,22 @@ def _values_for_rank(self):
             )
         return values
 
-    def ravel(self, order="C"):
+    def ravel(self, order: str = "C") -> "Categorical":
         """
-        Return a flattened (numpy) array.
+        Return a view on self.
 
         For internal compatibility with numpy arrays.
 
+        Parameters
+        ----------
+        order : str, default "C"
+            Parameter is ignored, is for compat with ndarray signature.
+
         Returns
         -------
-        numpy.array
+        Categorical
         """
-        warn(
-            "Categorical.ravel will return a Categorical object instead "
-            "of an ndarray in a future version.",
-            FutureWarning,
-            stacklevel=2,
-        )
-        return np.array(self)
+        return self.view()
```

Comment:
Not super familiar with this change - was there a consensus on this being the return value or does it need discussion?

Reply:
Yes, in #27199

Reply:
this is fine, though original we were returning self (and not a copy)?

Reply:
definitely not a copy, im not sure if it matters if we return `self` vs `self.view`

Reply:
this is a shallow copy (a new object)

```
   def view(self, dtype=None):
        if dtype is not None:
            raise NotImplementedError(dtype)
        return self._constructor(values=self._codes, dtype=self.dtype, fastpath=True)

```

but suppose its ok, can you validate that it is a new object in the tests?

Reply:
> can you validate that it is a new object in the tests?

After adding this to the test_ravel in the extensions tests, it looks like all the other EAs we test return self (because that is was the base class does so ill change this to just use the base class implementation and match the others.